src/views/Provisioners/ViewWorkers/index.jsx

```diff
@@ -0,0 +1,206 @@
+            'pageInfo',
+            pageInfo
+          )
+        );
+      },
+    });
+  };
+
+  handleFilterChange = ({ target }) => {
+    const {
+      data: { refetch },
+      match: {
+        params: { provisionerId, workerType },
+      },
+    } = this.props;
+    const quarantinedOpts =
+      target.value === 'Quarantined' ? { quarantined: true } : null;
+
+    this.setState({ filterBy: target.value });
+
+    refetch({
+      provisionerId,
+      workerType,
+      workersConnection: {
+        limit: VIEW_WORKERS_PAGE_SIZE,
+      },
+      ...quarantinedOpts,
+    });
+  };
+
+  // TODO: Handle action request
+  handleActionClick() {}
+
+  renderActions = () => {
+    const {
+      classes,
+      data: {
+        workerType: { actions },
+      },
+    } = this.props;
+
+    return actions.length
+      ? actions.map(action => (
+          <Tooltip
+            enterDelay={300}
+            key={action.title}
+            id={`${action.title}-tooltip`}
+            title={action.description}>
+            <Button
+              onClick={() => this.handleActionClick(action)}
+              className={classes.actionButton}
+              size="small"
+              variant="raised">
+              {action.title}
+            </Button>
+          </Tooltip>
+        ))
+      : 'n/a';
+  };
+
+  render() {
+    const { filterBy } = this.state;
+    const {
+      classes,
+      user,
+      onSignIn,
+      onSignOut,
+      match: {
+        params: { provisionerId, workerType },
+      },
+      data: { loading, error, workers },
+    } = this.props;
+
+    return (
+      <Dashboard
+        title="View Workers"
+        user={user}
+        onSignIn={onSignIn}
+        onSignOut={onSignOut}>
+        <Fragment>
+          {!workers && loading && <Spinner loading />}
+          {error && error.graphQLErrors && <ErrorPanel error={error} />}
+          {workers && (
+            <Fragment>
+              <Card>
+                <CardContent className={classes.cardContent}>
+                  <List>
+                    <ListItem>
+                      <Dropdown
+                        disabled={loading}
+                        onChange={this.handleFilterChange}
+                        value={filterBy || ''}
+                        inputLabel="Filter By">
+                        <MenuItem value="">
+                          <em>None</em>
+                        </MenuItem>
+                        <MenuItem value="Quarantined">Quarantined</MenuItem>
+                      </Dropdown>
+                    </ListItem>
+                    <ListItem>
+                      <ListItemText
+                        primary="Actions"
+                        secondary={this.renderActions()}
+                      />
+                    </ListItem>
+                  </List>
+                </CardContent>
+              </Card>
+              <br />
+              {loading ? (
```

Comment:
Why the loading check here? The connection table already has a loading indicator.